huaweicloud/services/oms/resource_huaweicloud_oms_migration_task_group.go

```diff
@@ -0,0 +1,835 @@
+	if err != nil {
+		return nil, err
+	}
+
+	sourceCdn, err := buildSourceCdnOpts(d.Get("source_cdn").([]interface{}))
+	if err != nil {
+		return nil, err
+	}
+
+	var migrateSinceOpt *int64
+	if v, ok := d.GetOk("migrate_since"); ok {
+		migrateSince, err := utils.FormatUTCTimeStamp(v.(string))
+		if err != nil {
+			return nil, err
+		}
+		migrateSinceOpt = &migrateSince
+	}
+
+	createOpts := &oms.CreateTaskGroupReq{
+		SrcNode:                     buildTaskGroupSrcNodeOpts(d.Get("source_object").([]interface{})),
+		Description:                 utils.StringIgnoreEmpty(d.Get("description").(string)),
+		DstNode:                     dstNodeOpts,
+		EnableFailedObjectRecording: utils.Bool(d.Get("enable_failed_object_recording").(bool)),
+		EnableKms:                   d.Get("enable_kms").(bool),
+		TaskType:                    &taskType,
+		BandwidthPolicy:             buildBandwidthPolicyOpts(d.Get("bandwidth_policy").([]interface{})),
+		SourceCdn:                   sourceCdn,
+		MigrateSince:                migrateSinceOpt,
+		EnableRequesterPays:         utils.Bool(d.Get("enable_requester_pays").(bool)),
+	}
+
+	if v, ok := d.GetOk("object_overwrite_mode"); ok {
+		var objectOverwriteMode oms.CreateTaskGroupReqObjectOverwriteMode
+		if err := objectOverwriteMode.UnmarshalJSON([]byte(v.(string))); err != nil {
+			return nil, fmt.Errorf("error parsing the argument object_overwrite_mode: %s", err)
+		}
+		createOpts.ObjectOverwriteMode = &objectOverwriteMode
+	}
+
+	if v, ok := d.GetOk("consistency_check"); ok {
+		var consistencyCheck oms.CreateTaskGroupReqConsistencyCheck
+		if err := consistencyCheck.UnmarshalJSON([]byte(v.(string))); err != nil {
+			return nil, fmt.Errorf("error parsing the argument consistency_check: %s", err)
+		}
+		createOpts.ConsistencyCheck = &consistencyCheck
+	}
+
+	return createOpts, nil
+}
+
+type TaskGroupActionConfig struct {
+	Action  string
+	Ctx     context.Context
+	Conf    *config.Config
+	GroupID string
+}
+
+func resourceMigrationTaskGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.HcOmsV2Client(conf.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating OMS client: %s", err)
+	}
+	createOpts, err := buildTaskGroupCreateOpts(conf, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp, err := client.CreateTaskGroup(&oms.CreateTaskGroupRequest{Body: createOpts})
+	if err != nil {
+		return diag.Errorf("error creating OMS migration task group: %s", err)
+	}
+	if resp.GroupId == nil {
+		return diag.Errorf("unable to find the task group ID")
+	}
+	groupID := *resp.GroupId
+	d.SetId(groupID)
+
+	err = waitForTaskGroupStartedOrCompleted(ctx, client, groupID, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.Errorf("error waiting for task group (%s) started or completed: %s", groupID, err)
+	}
+
+	if action, ok := d.GetOk("action"); ok {
```

Comment:
looks the value of action can only  be stop here, how about this
```suggestion
	if action, ok := d.GetOk("action"); ok && action.(string) == "stop" {
```